### ISSUE TITLE
`.Request` is not a function in Cloudflare Workers

### ISSUE THREAD

**Issue by ben-xD** [ably/ably-js#760]
When using Ably in Cloudflare Workers, the following errors are shown when `Ably.Rest.auth.createTokenRequest` or `Ably.Rest.auth.requestToken` is called. These don't happen in pure WebWorker environments and only in Cloudflare Workers. There was a separate [issue](https://github.com/ably/ably-js/issues/751) for WebWorkers being resolved by [this PR](https://github.com/ably/ably-js/pull/756).

![Screenshot 2021-05-17 at 10 37 40](https://user-images.githubusercontent.com/24711048/118467680-e8941f00-b6fb-11eb-8244-5f5adfc42a3d.png)

The error seems to be caused by calling `Http.Request` in  [line 136 of `browser/lib/util/https.js`](https://github.com/ably/ably-js/blob/eb5fdfaba72e0d5f1ea18096cedff537f828ee56/browser/lib/util/http.js#L136). However, why it **only** happens in Cloudflare Workers (and not Web Workers in general), I don't know. I personally am not blocked by this bug/ issue because I stopped using Cloudflare Workers because the debugging experience was not good. Firebase functions can be debugged with breakpoints in WebStorm, and it's been great. Though it might be good to support users who wish to use Ably on Cloudflare Workers. 

## Reproducing it
- Create a free cloudflare account if you don't have one. You don't need to do anything else, you automatically get given a workers.dev domain: e.g. `abcd.workers.dev`. You can change it if you want.
- Follow [getting started guide](https://github.com/ben-xD/ably-cloudflare-workers/blob/master/README.md#getting-started) from [this repo](https://github.com/ben-xD/ably-cloudflare-workers).
- Notice the error shown above in the terminal output

## A small bit of detail about Cloudflare Workers
 It configures webpack target to be webworker, but it also uses the **Service Worker** APIs to incept network requests/ act as a server/ serverless function. The reason they do this instead of running NodeJS is cost, performance and scalability. There is no cold start issue like normal serverless, much lower memory footprint. More about that [here](https://www.youtube.com/watch?v=HK04UxENH10). So perhaps it is because we do not support running in "service worker", though this is _just a guess_.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-853) by [Unito](https://www.unito.io)

**Comment by ben-xD** [comment]
This was fixed by https://github.com/ably/ably-js/pull/756, but further issues with Cloudflare workers are present here: https://github.com/ably/ably-js/issues/766

**Comment by ben-xD** [comment]
The error can be reproduced [here](https://github.com/ben-xD/ably-cloudflare-workers) on the main branch. I've updated it to use Ably's webworker webpack output. 

This is error is probably related to the other issue running in Cloudflare Workers: https://github.com/ably/ably-js/issues/766

**Comment by ben-xD** [comment]
I just got tokenRequests generated in Cloudflare workers, and I saw the token get generated in `wrangler tail`, but it didn't respond to my browser/ client because it timed out (Im on the free plan, the free tier has **10ms** [execution time limit](https://developers.cloudflare.com/workers/platform/limits#worker-limits)). It did show my console.log output though (probably because timing out in cloudflare doesn't mean execution stops immediately), so creating tokens **worked** in cloudflare.

I did the work in the main branch of this repo: https://github.com/ben-xD/ably-cloudflare-workers

So I am closing this. :)
